modules/mbedtls/mbedtls_init/mbedtls_init.h

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2022 Qualcomm Innovation Center, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MBEDTLS_INIT_H
+#define MBEDTLS_INIT_H
+
+/* This should be called by platforms that do not wish to
+ * have mbedtls initialised during kernel startup
+ */
+int mbedtls_init(const struct device *device);
```

Comment:
Exposing `const struct device *device` does not seem right. I think we should have dedicated function like `zephyr_mbedtls_heap_init(void)` or something similar called from static `mbedtls_init(const struct device *device)` when proper Kconfig option is selected.

Reply:
let me go have a noodle, thanks